lib/protobuf/decoder.ex

```diff
@@ -133,8 +133,24 @@ defmodule Protobuf.Decoder do
   end
 
   defdecoderp decode_delimited(field_number, message, props) do
-    <<bytes::bytes-size(value), rest::bits>> = rest
-    handle_value(rest, field_number, wire_delimited(), bytes, message, props)
+    bytes_remaining = byte_size(rest)
+
+    if value <= bytes_remaining do
+      <<bytes::bytes-size(value), rest::bits>> = rest
+      handle_value(rest, field_number, wire_delimited(), bytes, message, props)
+    else
+      field =
+        case props.field_props do
+          %{^field_number => %{name_atom: field_name}} -> "field #{field_name}"
+          _ -> "field_number #{field_number}"
+        end
+
+      msg =
+        "insufficient data decoding #{field}, " <>
+          "expected #{inspect(rest)} to be at least #{value} bytes"
```

Comment:
I believe the size will be more valuable for debugging than the actual contents. So I'd suggest something like:

```suggestion
          "expected it to be at least #{value} bytes but was #{bytes_remaining}"
```